### ISSUE TITLE
Add `--statistics` for an overview (like ruff)

### ISSUE THREAD

**Issue by Julian-J-S** [astral-sh/ty#686]
ruff has: `ruff check --statistics` which can be very useful.

I would love to see the same for ty.

Integration ruff / ty into a big repo and getting 10k+ "errors" is hard to manage.
Getting some idea of how many different problems and counts would be very helpful :)

**Comment by sharkdp** [comment]
Not exactly what you want, but something that you can use as a workaround in the meantime:

The [`ecosystem-analyzer`](https://github.com/astral-sh/ecosystem-analyzer/) program can parse ty's "concise" diagnostic output. You can use the `parse-diagnostics` subcommand to create structured diagnostic information, and then use `generate-report` to create a single HTML page overview of all diagnostics. This includes statistics (somewhat hidden in the filter-dropdowns), and allows you to browse different categories of errors easily:

```bash
alias ecosystem-analyzer='uvx --from "git+https://github.com/astral-sh/ecosystem-analyzer" ecosystem-analyzer'

uvx ty check --output-format=concise | ecosystem-analyzer parse-diagnostics
ecosystem-analyzer generate-report parsed-diagnostics.json 
```

If you supply `--project-location` and `--commit` to `parse-diagnostics`, you'll also get clickable GitHub links to the source code locations.

**Comment by spaceone** [comment]
I would also love to see it.

**Comment by Sai-Suraj-27** [comment]
Yes, this would be very helpful. Really like the  ruff's `--statistics` flag. Right now using Pyrefly's [`--summarize-errors`](https://pyrefly.org/en/docs/installation/#install) flag.

**Comment by MarcoGorelli** [comment]
> Yes, this would be very helpful. Really like the ruff's `--statistics` flag. Right now using Pyrefly's [`--summarize-errors`](https://pyrefly.org/en/docs/installation/#install) flag.

There's also the equivalent of `pyrefly report` / `pyright --verifytypes`

**Comment by CompRhys** [comment]
bump +1 would find this useful
